src/stdlib_linalg.fypp

```diff
@@ -92,6 +93,21 @@ module stdlib_linalg
     #:endfor
   end interface outer_product
 
+  ! repeat matrix (of 2d array)
+  interface repmat
+    !! version: experimental
+    !!
+    !! Creates large matrices from a small array, `repmat()` repeats the given values of the array to create the large matrix.
+    !! ([Specification](../page/specs/stdlib_linalg.html#
+    !! repmat-creates-large-matrices-from-a-small-array))
```

Comment:
What is the difference with the intrinsic `spread` (i.e. "replicates a source array ncopies times along a specified dimension")?

Reply:
> What is the difference with the intrinsic `spread` (i.e. "replicates a source array ncopies times along a specified dimension")?

"replicates a source **ARRAY** ncopies times along a specified dimension", this doesn't quite work for matrices (rank-2) as you will need to do some complicated trickery to reshape it.

Reply:
Actually ,Fortran's intrinsic array functions have some strange optimizations. It may generate temporary array and `copy`. And ifort will be slow if you do some elemental-wise array operation.